include/deal.II/base/parameter_handler.h

```diff
@@ -125,6 +126,104 @@ namespace Patterns
+     * int a = 5;
+     * Patterns::Integer p;
+     *
+     * std::string s = p.to_string(a); // s = "5"
+     * @endcode
+     */
+    template<class T>
+    std::string to_string(const T &value) const;
+
+
+    /**
+     * Translate the content of the string matching this pattern, into a
+     * value and store it in @p value.  If the type T is not
+     * compatible with the pattern, an excepion ExcIncompatibleType is
+     * thrown.
+     *
+     * Internally, this function wraps a pointer to @p value in a
+     * boost::any object and calls the virtual function string_to_any(). Derived
+     * patterns should implement string_to_any for this function to work properly.
+     *
+     * An example usage is here:
+     *
+     * @code
+     * int a = 0;
+     * Patterns::Integer p;
+     *
+     * p.to_value("5", a);
+     *
+     * std::cout << a << std::endl; // This prints 5.
+     * @endcode
+     */
+    template<class T>
+    void to_value(const std::string &s, T &value) const;
+
+
+    /**
+     * An incompatible type was encountered when trying to access
+     * a pointer contained in a boost::any object
+     *
+     * @ingroup Exceptions
+     */
+    DeclException2 (ExcIncompatibleType,
+                    boost::any &, PatternBase &,
+                    << "The type " << arg1.type().name()
+                    << " is not compatible with the pattern "
+                    << arg2.description(Text));
+
+  protected:
+    /**
+     * Convert a string to a value, and store its content in the type
+     * pointed to by a boost::any object.
+     *
+     * Derived patterns can use this function to convert a matching string
+     * to a compatible type, stored as a boost::any object.
+     *
+     * This class expects the boost::any object to store a pointer to a
+     * value type that is compatible with this pattern. If this is not the case,
+     * an exception ExcIncompatibleType is thrown.
+     */
+    virtual void string_to_any(const std::string &s, boost::any &v) const;
+
+    /**
+     * Convert a compatible value type to a string. The value type should
+     * be stored as a pointer in the boost::any object.
+     *
+     * Derived patterns can use this function to convert a compatible type
+     * to a string matching this pattern. A snippet implementation taken from
+     * Patterns::Integer is given below:
+     *
+     * @code
+     * std::string Integer::any_to_string(const boost::any &v) const {
+     *  if (v.type() == typeid(const int *))
+     *    return (std::to_string(*(boost::any_cast<const int *>(v))));
+     *  else {
+     *    // Handle errors here
+     *    return std::string();
+     *  }
+     * }
+     * @endcode
+     *
+     * This class expects the boost::any object to store a pointer to a
+     * value type that is compatible with this pattern.  If this is not the
+     * case, an exception ExcIncompatibleType is thrown.
+     */
+    virtual std::string any_to_string(const boost::any &v) const;
```

Comment:
it should really be a `=0`, but I did not want to implement it for all Patterns we have. For example I have no way of dealing with `Patterns::File`, so the default implementation is needed for those cases.